api/src/main/java/io/jsonwebtoken/SignatureAlgorithm.java

```diff
@@ -576,8 +575,16 @@ public static SignatureAlgorithm forSigningKey(Key key) throws InvalidKeyExcepti
         if (key instanceof SecretKey) {
 
             SecretKey secretKey = (SecretKey)key;
+            String secretKeyAlg = secretKey.getAlgorithm();
```

Comment:
Might want an assertion here to ensure that `secretKey.getAlgorithm()` always returns a non-null/non-empty String value, since without it, the added line 582 below would cause a null-pointer exception.